### PR TITLE
feat: create_repo, add_repo_to_installation func for github system

### PR DESCRIPTION
**Pull request by huydinhle**
Features: Support creating GH repo, and adding GH repo to installation

* Add create_repo() function
* Add Add_repo_to_installation() function

This added two functions into github system to create a github repo, and adding a github repo to a github installtion/app.

These two functions can be useful to create a workflow that create a brand new repo for your organization then immediately add that repo to a github installation (slack bot for example)

Example:
```
workflows:
  github_create_repo:
    steps:
      - call_function: github.createRepo
        with:
          org: honeyscience
      - call_function: github.addRepoToInstallation
        with:
          org: honeyscience
          repoid: '{{ .ctx.repoid }}'
```

### CODE REVIEW

File: github.yaml [hunk 1]
```diff
@@ -58,6 +58,31 @@ systems:
...
+        parameters:
+          URL: 'https://api.github.com/orgs/{{ .ctx.org }}/repos'
+          method: POST
+          content:
+            name: '{{ .ctx.name }}'
+            private: true
+        export:
+          repoid: '{{ .data.json.id | int | toString }}'
+
+      addRepoToInstallation:
```
Comment (wyardley) [inline github.yaml]: maybe we could do this for Renovate and semantic release commit checker bots too?

File: github.yaml [hunk 2]
```diff
@@ -58,6 +58,31 @@ systems:
...
+          system: github
+          function: api
+        parameters:
+          URL: 'https://api.github.com/orgs/{{ .ctx.org }}/repos'
+          method: POST
+          content:
+            name: '{{ .ctx.name }}'
+            private: true
+        export:
+          repoid: '{{ .data.json.id | int | toString }}'
```
Comment (Charles546) [inline github.yaml]: Do we really need the `| int | toString`? I prefer using `$data.json.id`, this avoids the a go template rendering.

File: github.yaml [hunk 3]
```diff
@@ -58,6 +58,31 @@ systems:
...
+
+      addRepoToInstallation:
+        target:
+          system: github
+          function: api
+        parameters:
+          URL: 'https://api.github.com/user/installations/{{ .ctx.installationid | int | toString }}/repositories/{{ .ctx.repoid }}'
```
Comment (Charles546) [inline github.yaml]: Same applies

File: github.yaml [hunk 4]
```diff
@@ -58,6 +58,31 @@ systems:
...
+      createRepo:
+        target:
+          system: github
+          function: api
+        parameters:
+          URL: 'https://api.github.com/orgs/{{ .ctx.org }}/repos'
+          method: POST
+          content:
+            name: '{{ .ctx.name }}'
+            private: true
```
Comment (Charles546) [inline github.yaml]: Maybe make this a parameter

Comment (huydinhle) [reply github.yaml]: sounds good

File: github.yaml (hunk 2 above)
Comment (huydinhle) [reply github.yaml]: I tried that but It came out float for something which is something like 2.7214 10^7

Comment (huydinhle) [reply github.yaml]: repoid:2.33673173e+08

File: github.yaml (hunk 3 above)
Comment (huydinhle) [reply github.yaml]: done!

File: github.yaml (hunk 2 above)
Comment (huydinhle) [reply github.yaml]: I removed the toString part but I think we need to convert it to int because the repoid wasn't an int

**Comment by huydinhle** [comment]
@Charles546 addressed all the comments

**Pull request merged**